packages/react/src/query/query.ts

```diff
@@ -0,0 +1,58 @@
+import { ErrorCode, EvaluationDetails, FlagMetadata, FlagValue, StandardResolutionReasons } from '@openfeature/core';
+
+export interface FlagQuery<T extends FlagValue = FlagValue> {
```

Comment:
I tried to strike a balance here between adding too much and also being useful.

It mostly adds everything on `EvaluationDetails`, plus some stuff more useful to React usecases and this hook (type, since hook is generic).